test/pylib/scylla_cluster.py

```diff
@@ -291,19 +292,17 @@ async def cql_is_up(self) -> bool:
             caslog.setLevel(oldlevel)
         # Any other exception may indicate a problem, and is passed to the caller.
 
-    async def rest_api_is_up(self) -> bool:
+    async def get_host_id(self, api: ScyllaRESTAPIClient) -> bool:
         """Test that the Scylla REST API is serving. Can be used as a
         checker function at start up."""
         try:
-            async with aiohttp.ClientSession() as session:
-                url = f"http://{self.hostname}:10000/"
-                async with session.get(url):
-                    return True
-        except aiohttp.ClientConnectionError:
+            self.host_id = await api.get_host_id(self.hostname)
+            return True
+        except (aiohttp.ClientConnectionError, RuntimeError):
```

Comment:
Doesn't `RuntimeError` indicate a problem? Shouldn't we propagate it?

Reply:
Fixed in https://github.com/scylladb/scylladb/commit/0ebda403004c746f8a0019098e9125c160433847 https://github.com/scylladb/scylladb/compare/6635e6163463bf74df6826ee15735d79e6493578..0ebda403004c746f8a0019098e9125c160433847

Now `aiohttp.request` raises (its own Exceptions) and we don't do manual re-raising with `RuntimeError`s.